features/src/autogluon/features/generators/datetime.py

```diff
@@ -47,9 +47,15 @@ def _generate_features_datetime(self, X: DataFrame) -> DataFrame:
             # TODO: Be aware: When converted to float32 by downstream models, the seconds value will be up to 3 seconds off the true time due to rounding error. If seconds matter, find a separate way to generate (Possibly subtract smallest datetime from all values).
             # TODO: could also return an extra boolean column is_nan which could provide predictive signal.
             X_datetime[datetime_feature] = pd.to_datetime(X[datetime_feature], errors='coerce').fillna(self._fillna_map[datetime_feature])
-            X_datetime[datetime_feature] = pd.to_numeric(X_datetime[datetime_feature])  # TODO: Use actual date info
             # X_datetime[datetime_feature] = pd.to_timedelta(X_datetime[datetime_feature]).dt.total_seconds()
-            # TODO: Add fastai date features
+            # Parse the date into lots of derived fields.
+            # Most of the pandas Series.dt properties are here, a few are omitted (e.g. is_month_start) if they can be inferred
```

Comment:
Another is `is_leap_year`, unsure how useful that is on most datasets. I think it's probably not worthwhile. `quarter` can be inferred from month quite easily, so it can also be cut.

Reply:
To further reduce features, the following can also be cut:
- dayofyear (inferrable from month+day)
- is_month_end (hard to imagine the use case except accounting users, and a tree could build this up from month+day if it really helped)

This just leaves us with defaults of ['year', 'month', 'day', 'dayofweek']

I'm tempted to also tackle the issue where pd.Series['20200101', '20210302'] is not recognised as a datetime.  My personal dataset hits that problem.  Don't worry, I'll do it with a subsample and probably initially some heuristics (8 characters, starts with '1' or '2') because pd.to_datetime is slow because it's powerful and supports a huge range of date formats ('12/31/06', '31-12-1980', 'Jan-87', '2020-09-07', ...).

Reply:
Sounds good! I think the 4 you mention make sense as defaults.